browser/src/Editor/NeovimEditor.tsx

```diff
@@ -172,7 +172,21 @@ export class NeovimEditor extends Editor implements IEditor {
             commandManager.executeCommand(command)
         })
 
-        this._neovimInstance.on("event", (eventName: string, evt: any) => this._onVimEvent(eventName, evt))
+        this._neovimInstance.on("event", (eventName: string, evt: any) => {
+            const current = evt.current || evt
+            const buf = this._bufferManager.updateBufferFromEvent(current)
+            switch (eventName) {
+                case "BufEnter":
```

Comment:
As opposed to using the switch here, could we hook the `_onBufEnter` event by using the strongly-typed event, like:

`this._neovimInstance.autoCommands.onBufEnter.subscribe((evt) => this._onBufEnter)`? LIikewise for `BufWipeout` and `BufWritePost` - this will help ensure we can leverage the typed events for these.

This way, the only stuff happening in here is the stuff we were doing on every event anyway - the `this._bufferManager.updateBufferFromEvent`, the `UI.Actions.setWindowCursor`, and the syntax highlighter change. And later, in another change, we can refactor those out, and remove this untyped event completely.